NodeAnimations.md

```diff
@@ -0,0 +1,208 @@
+
+### Path Following with Camera Tracking
+To have a camera following a path and tracking a target we will be combining the things learned in the previous sections.
+
+Create a new scene and delete the default cube.
+
+Add three empty objects (3 x `Shift-A->Empty->Plain Axes`) and name them as follows: *Tracked*, *CameraBase* and *CameraArm*.
+
+Parent the empty *CameraArm* to *CameraBase* so that the arm is a child of the base, and then parent the camera to *CameraArm*.
+
+Clear the camera location and rotation by pressing `Alt-G`, `Alt-R`.
+
+Add a Monkey mesh with `Shift-A->Mesh->Monkey` to have some object for the camera to look at.
+
+Then add a BezierCircle to have a path to follow, scale it and name it *CameraPath*.
+
+Select the *CameraBase* empty and add a constraint to follow the *CameraPath* bezier curve with `Forward:` -X and `Up:` Z (also check `Follow Curve` and `Fixed Position`).
+
+Select the *CameraArm* empty and move it upwards so that the camera is looking the object from a vantage point and not the floor.
+
+Then select the camera and add a constraint of type `Locked Track`, with `Target:` *Tracked*, `To:` -Z and `Lock:` X.
+
+The purpose of having the camera track the *Tracked* empty object instead of the Monkey mesh is that now it is not locked to the center of the object.
+
+If the Monkey object is going to move as well, then the *Tracked* empty object can be made a child of the Monkey object or use more empty objects to move the monkey.
+
+It is a good general rule to separate movements into indepentent nodes as was done with *CameraBase* and *CameraArm* that are generally used to separate the yaw and pitch of the camera.
+
+Now, select *CameraBase* and repeat the process as in the previous tutorial to have the *CameraBase* empty follow the circular path, in the process the camera will keep on tracking the *Tracked* empty object.
+
+It should be noted that Ogre has a `setAutoTracking()` function for the SceneNodes, which would be a preferable way to track objects.
+
+Please consult the manual for more details: https://ogrecave.github.io/ogre/api/latest/class_ogre_1_1_scene_node.html#ae4c8588895d3623bbe0007ea157af1a4
+
+## Using Node Animations in you Ogre app
+If you are using DotScene to load the .scene into your Ogre app, by default the animations won't play because they are disabled by default.
+
+Besides enabling the animations it is also required to `addTime()` to their *AnimationStates*.
+
+```
+int main()
+{
+	...
+    for (auto animationState : mSceneMgr->getAnimationStates())
+    {
+		std::cout << "animationState: " << animationState.second->getAnimationName() << std::endl;
+		animationState.second->setEnabled(true);
+		animationState.second->setLoop(true);
+    }
+}
+
+bool MyOgreApp::frameRenderingQueued(const Ogre::FrameEvent& evt)
```

Comment:
this is not the best thing to do. Instead this should be used
```cpp
        auto& controllerMgr = ControllerManager::getSingleton();
        controllerMgr.createFrameTimePassthroughController(AnimationStateControllerValue::create(animState, true));
```

Reply:
How can you have this massive API loaded in your head? LoL

I wasn't aware of that class, will look into it.